### DISCUSSION TITLE
Support for services like google drive for postgresql backups

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/docker-duplicity#644] [Q&A]
I am using the docker-duplicity-postgres:latest image to back up a project made with doodba. In the documentation they suggest Amazon S3, but I want to try Google Drive. These images support other providers. In the copier questions, it did not ask me for any data for gdocs

**Comment by Daemo00** [comment]
Hi,
I was trying to do the same and was very sad when I found this unanswered discussion.
After some trial-and-error I managed to make it work, there might be some other ways, but this is how I did it:

1. Create a project in the Google developers console, maybe you already have it for other APIs used in the Odoo you are deploying
2. In https://console.cloud.google.com/iam-admin/serviceaccounts, create a service account in your project
3. Create and download the JSON key for your service account and place it in the **.docker** folder, with name **backup-key.json**.
    Note that this must <ins>not</ins> be version controlled with `git`, its security is as critical as any other file in **.docker**!
5. In Google drive, share a folder to the service account
4. Follow https://duplicity.nongnu.org/vers8/duplicity.1.html#sect27 to build the URL
    > gdrive://\<service account>@\<project>.iam.gserviceaccount.com/\<name of folder to create in shared folder>?myDriveFolderID=\<ID of the shared folder>

    The ID of a folder in Google drive is simply the last part of its URL
5. This URL is the answer to the question
    > If you want to use any other backend, supply any URL supported by Duplicity

    (in https://github.com/Tecnativa/doodba-copier-template/blob/c0a193f059829a789d963128b294ffc65761ef2e/copier.yml#L537C5-L537C80)
6. In the **backup** service of your **prod.yaml**, add
    ```yaml
    volumes:
      - .docker:/root/.docker:ro
    ```
7. In **.docker/backup.env**, add
    ```sh
    GOOGLE_SERVICE_JSON_FILE="/root/.docker/backup-key.json"
    ```
    It is needed by **Duplicity** as explained in https://duplicity.nongnu.org/vers8/duplicity.1.html#sect27
8. Build as explained in https://github.com/Tecnativa/doodba-copier-template/blame/c0a193f059829a789d963128b294ffc65761ef2e/docs/daily-usage.md#L235:
    ```sh
    docker compose -f prod.yaml up -d
    ```
9. Try it as explained in https://github.com/Tecnativa/doodba-copier-template/blame/c0a193f059829a789d963128b294ffc65761ef2e/docs/daily-usage.md#L259:
    ```sh
    docker compose exec backup /etc/periodic/daily/jobrunner
    ```
    Actually, the command that I have to use is:
    ```sh
    docker-compose -f prod.yaml exec backup /etc/periodic/daily/jobrunner
    ```
    but I suppose that's just me

And that's it, it will work!
I'm joking, it won't work at the first try of course, because you or me will have done some mistakes in the process, but it should get you close enough.

**Reply by josep-tecnativa** [reply]
Could you please, share which error do you get?

**Reply by Daemo00** [reply]
I'm not getting any error, with the above procedure it works
